activerecord/test/cases/relations_test.rb

```diff
@@ -1349,6 +1349,23 @@ def test_find_or_create_by!
     assert_raises(ActiveRecord::RecordInvalid) { Bird.find_or_create_by!(color: "green") }
   end
 
+  def test_create_or_find_by
+    assert_nil Subscriber.find_by(nick: "bob")
+
+    subscriber = Subscriber.create!(nick: "bob")
+
+    assert_equal subscriber, Subscriber.create_or_find_by(nick: "bob")
+    assert_not_equal subscriber, Subscriber.create_or_find_by(nick: "cat")
+  end
+
+  def test_create_or_find_by_with_non_unique_attributes
+    Subscriber.create!(nick: "bob", name: "the builder")
+
+    assert_raises(ActiveRecord::RecordNotFound) do
+      Subscriber.create_or_find_by(nick: "bob", name: "the cat")
+    end
+  end
+
```

Comment:
I think this is missing a test with `create_with` (to change the behavior of the `create`) and with `where` (to change the behavior of `find!`

I'm not sure if those method should be used with `create_or_find_by` but right now they can and I have no idea of the effect they would have, so better to test them to make sure it is what we expect.

Reply:
I don't see how that differs? `create_or_find_by` uses the same flow as `find_or_create_by`, except instead of an `||` we're using a `rescue`. But the flow is the same.

Reply:
I mean, it's not that we can't test it, just that I don't think it teaches us anything interesting. The test would be identical to the one for `test_find_or_create_by_with_create_with`.

Reply:
If it is the same behavior we want so I guess it is fine being tested in that test